File: go-service/models/goods_received_note.go
package models

import "time"

// GoodsReceivedNote represents a GRN for received goods
type GoodsReceivedNote struct {
	ID               int       `db:"Id" json:"id"`
	GRNNumber        string    `db:"GRNNumber" json:"grnNumber"`
	PurchaseOrderID  int       `db:"PurchaseOrderId" json:"purchaseOrderId"`
	ReceivedByUserID int       `db:"ReceivedByUserId" json:"receivedByUserId"`
	ReceivedDate     time.Time `db:"ReceivedDate" json:"receivedDate"`
	Notes            *string   `db:"Notes" json:"notes,omitempty"`
	Status           string    `db:"Status" json:"status"`
	CreatedAt        time.Time `db:"CreatedAt" json:"createdAt"`
}

// GoodsReceivedNoteItem represents an item in a GRN
type GoodsReceivedNoteItem struct {
	ID                  int       `db:"Id" json:"id"`
	GoodsReceivedNoteID int       `db:"GoodsReceivedNoteId" json:"goodsReceivedNoteId"`
	ProductID           int       `db:"ProductId" json:"productId"`
	QuantityOrdered     int       `db:"QuantityOrdered" json:"quantityOrdered"`
	QuantityReceived    int       `db:"QuantityReceived" json:"quantityReceived"`
	Remarks             *string   `db:"Remarks" json:"remarks,omitempty"`
	CreatedAt           time.Time `db:"CreatedAt" json:"createdAt"`
}

File: frontend/src/app/core/domain/goods-received-note/goods-received-note.ts
import { GoodsReceivedNoteStatus } from '../../enums/goods-received-note-status';
import { PurchaseOrder } from '../purchase-order/purchase-order';
import { User } from '../user/user';

/**
 * Represents a Goods Received Note Item
 */
export interface GoodsReceivedNoteItem {
  /** ID */
  id: number;

  /** Goods Received Note ID */
  goodsReceivedNoteId: number;

  /** Product ID */
  productId: number;

  /** Quantity Ordered */
  quantityOrdered: number;

  /** Quantity Received */
  quantityReceived: number;

  /** Remarks */
  remarks?: string | null;

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;
}

/**
 * Represents a Goods Received Note
 */
export interface GoodsReceivedNote {
  /** ID */
  id: number;

  /** GRN Number - unique identifier */
  grnNumber: string;

  /** Purchase Order ID - FK to PurchaseOrder */
  purchaseOrderId: number;

  /** Received By User ID - FK to User */
  receivedByUserId: number;

  /** Received Date */
  receivedDate: Date;

  /** Notes */
  notes?: string | null;

  /** Status */
  status: GoodsReceivedNoteStatus;

  /** Purchase Order - optional navigation property */
  purchaseOrder?: PurchaseOrder;

  /** Received By User */
  receivedByUser?: User;

  /** Items - GRN items */
  items?: GoodsReceivedNoteItem[];

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;
}
